src/twisted/test/proto_helpers.py

```diff
@@ -159,6 +162,7 @@ class StringTransport:
     """
 
     disconnecting = False
+    aborted = False
```

Comment:
`disconnecting` is set synchronously; `disconnected` is not set synchronously by `loseConnection`, but it _is_ set by `abortConnection`, because `abortConnection` actually causes the connection to go away.

Inasmuch as these existing flags have clear semantics, `disconnected` means "there's no transport any more", which is more or less what you're asking `abortConnection` to accomplish.